### ISSUE TITLE
Object detection example not working for custom transformation Flash version 0.6.0

### ISSUE THREAD

**Issue by Actis92** [PyTorchLightning/lightning-flash#1107]
Here is described how use the ObjectDetector Module, and how pass custom transformations. As you can see in the example it is suggested to define a dictionary, but using it with Flash 0.6.0 it didn't work.
But i obtain the following error:
File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/flash/core/data/io/input_transform.py", line 1112, in create_transform
    raise MisconfigurationException(f"The format for the transform isn't correct. Found {transform}")
pytorch_lightning.utilities.exceptions.MisconfigurationException: The format for the transform isn't correct. Found {'per_sample_transform': IceVisionTransformAdapter()}

because in the function create_transform in the file flash/core/data/io/input_transform.py not works with dictionary as input value. Someone knows what is the correct way to pass custom transformation with the latest release of Flash?

**Comment by ethanwharris** [comment]
Hi @Actis92 thanks for reporting this. We recently went through a refactor but some docs got left behind. We no longer support dictionary transforms and instead rely on the `InputTransform` class. The new version of that example would look like something this:
```
from dataclasses import dataclass
import albumentations as alb
from icevision.tfms import A
from flash import InputTransform
from flash.core.integrations.icevision.transforms import IceVisionTransformAdapter


@dataclass
class BrightnessContrastTransform(InputTransform):

    image_size: int = 128

    def per_sample_transform(self):
        return IceVisionTransformAdapter(
            [*A.aug_tfms(size=self.image_size), A.Normalize(), alb.RandomBrightnessContrast()]
        )


datamodule = ObjectDetectionData.from_coco(
    train_folder="data/coco128/images/train2017/",
    train_ann_file="data/coco128/annotations/instances_train2017.json",
    val_split=0.1,
    train_transform=BrightnessContrastTransform,
    transform_kwargs={"image_size": 128},
    batch_size=4,
)
```

**Comment by Actis92** [comment]
Hi @ethanwharris thank you for tell me how pass the transformation, but I continue to have problem with the example.
I'm trying this code:
`from dataclasses import dataclass
import albumentations as alb
import flash
from flash.core.data.utils import download_data
from flash.image import ObjectDetectionData, ObjectDetector
from icevision.tfms import A
from flash import InputTransform
from flash.core.integrations.icevision.transforms import IceVisionTransformAdapter


if __name__ == "__main__":


    download_data("https://github.com/zhiqwang/yolov5-rt-stack/releases/download/v0.3.0/coco128.zip", "data/")


    @dataclass
    class BrightnessContrastTransform(InputTransform):
        image_size: int = 128

        def per_sample_transform(self):
            return IceVisionTransformAdapter(
                [*A.aug_tfms(size=self.image_size), A.Normalize(), alb.RandomBrightnessContrast()]
            )


    datamodule = ObjectDetectionData.from_coco(
        train_folder="data/coco128/images/train2017/",
        train_ann_file="data/coco128/annotations/instances_train2017.json",
        val_split=0.1,
        train_transform=BrightnessContrastTransform,
        transform_kwargs={"image_size": 128},
        batch_size=4,
    )

    # 2. Build the task
    model = ObjectDetector(head="efficientdet", backbone="d0", num_classes=datamodule.num_classes,
                           image_size=128)

    # 3. Create the trainer and finetune the model
    trainer = flash.Trainer(max_epochs=1)
    trainer.finetune(model, datamodule=datamodule, strategy="freeze")`

But I obtain this error:

`File "/Users/lucaactisgrosso/Library/Application Support/JetBrains/PyCharmCE2021.1/scratches/scratch_33.py", line 42, in <module>
    trainer.finetune(model, datamodule=datamodule, strategy="freeze")
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/flash/core/trainer.py", line 181, in finetune
    return super().fit(model, train_dataloader, val_dataloaders, datamodule)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 741, in fit
    self._fit_impl, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 685, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 777, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 1199, in _run
    self._dispatch()
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 1279, in _dispatch
    self.training_type_plugin.start_training(self)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/plugins/training_type/training_type_plugin.py", line 202, in start_training
    self._results = trainer.run_stage()
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 1289, in run_stage
    return self._run_train()
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/flash/core/trainer.py", line 111, in _run_train
    self.fit_loop.run()
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/loops/base.py", line 145, in run
    self.advance(*args, **kwargs)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/loops/fit_loop.py", line 234, in advance
    self.epoch_loop.run(data_fetcher)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/loops/base.py", line 140, in run
    self.on_run_start(*args, **kwargs)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/loops/epoch/training_epoch_loop.py", line 141, in on_run_start
    self._dataloader_iter = _update_dataloader_iter(data_fetcher, self.batch_idx + 1)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/loops/utilities.py", line 121, in _update_dataloader_iter
    dataloader_iter = enumerate(data_fetcher, batch_idx)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/utilities/fetching.py", line 199, in __iter__
    self.prefetching(self.prefetch_batches)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/utilities/fetching.py", line 258, in prefetching
    self._fetch_next_batch()
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/utilities/fetching.py", line 300, in _fetch_next_batch
    batch = next(self.dataloader_iter)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/trainer/supporters.py", line 550, in __next__
    return self.request_next_batch(self.loader_iters)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/trainer/supporters.py", line 562, in request_next_batch
    return apply_to_collection(loader_iters, Iterator, next)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/pytorch_lightning/utilities/apply_func.py", line 95, in apply_to_collection
    return function(data, *args, **kwargs)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 521, in __next__
    data = self._next_data()
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 561, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 52, in fetch
    return self.collate_fn(data)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/flash/core/integrations/icevision/adapter.py", line 88, in _collate_fn
    [to_icevision_record({**sample, DataKeys.METADATA: m}) for sample, m in zip(samples, metadata)]
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/icevision/models/utils.py", line 88, in inner
    tupled_output, records = build_batch(records)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/icevision/models/utils.py", line 79, in inner
    return build_batch(records, **build_batch_kwargs)
  File "/Users/lucaactisgrosso/PycharmProjects/lit-tracking/venv/lib/python3.7/site-packages/icevision/models/ross/efficientdet/dataloaders.py", line 98, in build_train_batch
    batch_images = torch.stack(batch_images)
RuntimeError: stack expects each tensor to be equal size, but got [3, 480, 640] at entry 0 and [3, 427, 640] at entry 2`

As you can see in this screen all the images have different size, so it seems that it didn't apply the transformation
<img width="244" alt="Schermata 2022-01-11 alle 09 39 02" src="https://user-images.githubusercontent.com/46601193/148909024-a3f40792-e567-4c71-bb3b-45b234712f9c.png">

Let me know if I'm doing something wrong.
Thank you :)

**Comment by ethanwharris** [comment]
Ah @Actis92 that looks like a bug we fixed on master but didn't release yet. If you install flash with:
```
pip install 'git+https://github.com/PyTorchLightning/lightning-flash.git#egg=lightning-flash[image]'
```
Then it should work. We have quite a few bugs fixed on master so I'm hoping to do a bugfix release soon :smiley:

Hope that helps!

**Comment by Actis92** [comment]
Thank you very much @ethanwharris, with the version on branch master the example works correctly 😃 
Since I have done this tests I have opened a PR in order to fix the documentation, as you can see in #1110
